packages/core/src/Repo.ts

```diff
@@ -392,12 +365,20 @@ export class Repo<T = any> extends EventEmitter {
 
   // PRIVATE
 
+  /**
+   * Determines whether the repo has previously persisted data or not.
+   * @returns `true` if there is any stored data in the repo.
+   */
+  private async hasData() {
+    return this.storage.hasData()
+  }
+
   /**
    * Loads all the repo's snapshots into memory
    */
   private async loadSnapshotsFromDb() {
-    const snapshots = await this.database!.getAll('snapshots')
-    for (const { documentId, snapshot } of snapshots) {
+    for await (const cursor of this.storage.snapshots) {
+      const { documentId, snapshot } = cursor.value as SnapshotRecord
```

Comment:
We won't need this cast once we fix the type of `snapshots`

Reply:
👍 58de7e1ec3557b21262987bd681db3f7ad9ef74e